src/Numerics/DGMethods/DGModel.jl

```diff
@@ -719,6 +762,15 @@ function (dg::DGModel)(tendency, state_prognostic, _, t, α, β)
     # The synchronization here through a device event prevents CuArray based and
     # other default stream kernels from launching before the work scheduled in
     # this function is finished.
+    if dg.tendency_filter !== nothing
+        comp_stream = Filters.apply!(
```

Comment:
```suggestion
        comp_stream = Filters.apply_async!(
```

Reply:
Thanks @mwarusz but that still doesn't work :/

Reply:
Weird, works for me. Just to make sure: I had two separate suggestions, your latest has the first one but not this one. Is this what you wanted ?

Reply:
Oh, yes! Sorry! I had accidentally missed one of the two. Thanks! Just testing it now.